draft-ietf-httpbis-sfbis.md

```diff
@@ -839,7 +839,7 @@ NOTE: This algorithm parses both Integers ({{integer}}) and Decimals ({{decimal}
    1. If the final character of input_number is ".", fail parsing.
    2. If the number of characters after "." in input_number is greater than three, fail parsing.
    2. Parse input_number as a decimal number and let output_number be the product of the result and sign.
-0. Return output_number.
+0. Return output_number, type.
```

Comment:
There is an alternative framing here.

Change step 8 to be:

```
8. If type is "integer":
    1. Let output_number be an Integer that is the result of parsing input_number as an integer.
9. Otherwise:
    1. If the final character of input_number is ".", fail parsing.
    2. If the number of characters after "." in input_number is greater than three, fail parsing.
    3. Let output_number be an Decimal that is the result of parsing input_number as a decimal.
13. Multiply output_number by sign.
14. Return output_number.
```

Not that invasive, but you are saying that output_number has both a type and a value.

You could do that consistently as well, I guess, but this is one of the few places that the output of a function like this matters.

Reply:
That could work.